servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -287,14 +299,25 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge)
           .merge(
               toHash(merge.getFromRefName(), merge.getFromHash(), true).get(),
               BranchName.of(branchName),
-              toHash(hash, true));
+              toHash(hash, true),
+              updateMetaWithMergePropsFunc(MergeOp.MERGE, merge.getFromRefName()));
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(e.getMessage(), e);
     } catch (ReferenceConflictException e) {
       throw new NessieConflictException(e.getMessage(), e);
     }
   }
 
+  private BiFunction<Serializer<CommitMeta>, ByteString, ByteString> updateMetaWithMergePropsFunc(
```

Comment:
This should be moved into the enum. The enum's is actually not needed at all wrt to the comment below.